### ISSUE TITLE
Cleanup `duke/projects`

### ISSUE THREAD

**Issue by alexfoias** [neuropoly/data-management#131]
Here is a list of all folders:

```
ls -latd /Volumes/projects/* | head -n 100
drwx------  1 alex  staff   264 28 Sep 14:45 /Volumes/projects/ml_deep_unwrapping
drwx------  1 alex  staff   874  5 Sep 17:26 /Volumes/projects/ukbiobank
drwx------  1 alex  staff   908  9 Jul 15:52 /Volumes/projects/axondeepseg
drwx------  1 alex  staff  1384 19 Jun 18:29 /Volumes/projects/ivadomed
drwx------  1 alex  staff   432  7 Jun 11:18 /Volumes/projects/ukbiobank_SI_PP
drwx------  1 alex  staff   264  4 Jun 10:38 /Volumes/projects/archived
drwx------  1 alex  staff   264 22 Mar  2021 /Volumes/projects/spine_generic
drwx------  1 alex  staff   364  9 Mar  2021 /Volumes/projects/ml_vertebral_labeling
drwx------  1 alex  staff   264 24 Feb  2021 /Volumes/projects/shimming-toolbox
drwx------  1 alex  staff   264 25 Jan  2021 /Volumes/projects/connectome_random_walk_Parkinsons
drwx------  1 alex  staff   264 17 Dec  2020 /Volumes/projects/csf_deepseg
drwx------  1 alex  staff   432  3 Dec  2020 /Volumes/projects/ms_seg
drwx------  1 alex  staff   264 30 Nov  2020 /Volumes/projects/exvivo_mri_queensland
drwx------  1 alex  staff   330  8 Oct  2020 /Volumes/projects/ml_sci_prognosis
drwx------  1 alex  staff   500 13 Aug  2020 /Volumes/projects/long_ms_brain_spine
drwx------  1 alex  staff   534 14 May  2020 /Volumes/projects/ms_brain_spine
drwx------  1 alex  staff   296 25 Mar  2020 /Volumes/projects/stainless_microscopy
drwx------  1 alex  staff   500 29 Jan  2020 /Volumes/projects/RAMRI
drwx------  1 alex  staff   568 28 Nov  2019 /Volumes/projects/gm_challenge
drwx------  1 alex  staff   264 19 Oct  2019 /Volumes/projects/sct_deepseg_threshold
-rwx------  1 alex  staff    16  4 Sep  2019 /Volumes/projects/SyncToy_75f8bdd3-e1ea-4a35-8883-0b8787580c15.dat
drwx------  1 alex  staff   264  4 Sep  2019 /Volumes/projects/tract_clustering
drwx------  1 alex  staff   264  4 Sep  2019 /Volumes/projects/schwab_gmseg
drwx------  1 alex  staff  1214  4 Sep  2019 /Volumes/projects/ms_lesion_mapping
drwx------  1 alex  staff   976  4 Sep  2019 /Volumes/projects/ms_connectome
drwx------  1 alex  staff   264  4 Sep  2019 /Volumes/projects/mrihisto
drwx------  1 alex  staff   264  4 Sep  2019 /Volumes/projects/HeadFix
drwx------  1 alex  staff   264  4 Sep  2019 /Volumes/projects/generic_spine_procotol
drwx------  1 alex  staff   264  4 Sep  2019 /Volumes/projects/ex_vivo_qmri
drwx------  1 alex  staff   264  4 Sep  2019 /Volumes/projects/exvivo_duke
drwx------  1 alex  staff   264  4 Sep  2019 /Volumes/projects/CPZ
drwx------  1 alex  staff   264  4 Sep  2019 /Volumes/projects/connectome_random_walk_MS
drwx------  1 alex  staff   364  4 Sep  2019 /Volumes/projects/atlas_rat
drwx------  1 alex  staff   264 11 May  2018 /Volumes/projects/ml_nih_segmentation
drwx------  1 alex  staff   264  5 Oct  2017 /Volumes/projects/atlas_human
```
I suggest moving all the folders before 2020 to `duke/archives`.

@jcohenadad @kousu What do you think ?

**Comment by jcohenadad** [comment]
> I suggest moving all the folders before 2020 to duke/archives.

no: some projects are still active and should not be considered "archived"

**Comment by charleygros** [comment]
`/Volumes/projects/ivadomed` is active :-) thank you

**Comment by jcohenadad** [comment]
also, i see 2019/atlas_human but it should be under 2017

pls be careful with the dates...

**Comment by kousu** [comment]
We probably want to look at `atime`. Here's the alphabetically first files that haven't been used in 2 years (`+730` days), not counting things already in `archived/`:

```
u108545@joplin:~$ find  duke/projects/ -name archived -prune -o -atime +730 -type f | head
find: ‘duke/projects/.TemporaryItems’: Permission denied
duke/projects/archived
duke/projects/axondeepseg/20171120_AxonDeepSeg_paper_results/code/.DS_Store
duke/projects/axondeepseg/20171120_AxonDeepSeg_paper_results/code/as_display_label_axonseg.m
duke/projects/axondeepseg/20171120_AxonDeepSeg_paper_results/code/batch_axondeepseg.m
duke/projects/axondeepseg/20171120_AxonDeepSeg_paper_results/code/batch_deep.m
duke/projects/axondeepseg/20171120_AxonDeepSeg_paper_results/code/batch_metrics_axondeepseg.m
duke/projects/axondeepseg/20171120_AxonDeepSeg_paper_results/code/boxplot_TEM.m
duke/projects/axondeepseg/20171120_AxonDeepSeg_paper_results/code/dependencies/j_disp.m
duke/projects/axondeepseg/20171120_AxonDeepSeg_paper_results/code/dependencies/load_nii_data.m
duke/projects/axondeepseg/20171120_AxonDeepSeg_paper_results/code/dependencies/make_nii.m
```

there don't seem to be many files older than 3 years though:

```
u108545@joplin:~$ find  duke/projects/ -name archived -prune -o -atime +$((365*3)) -type f | head
find: ‘duke/projects/.TemporaryItems’: Permission denied
^C
```

**Comment by kousu** [comment]
I thought about this over the weekend and came up with a forumla:


```
rank = (lg(size)/30 + age/(900 days))/2
```

`duke/projects` has files/folders (very roughly) uniformly distributed in age between now and 3 years ago. There's gaps (between semesters?) and slightly more weight on very new _and very old_ (too old for anyone to remember to clean them up?). And they're distributed in size exponentially, since that's how just how files work -- many small files/folders, a few very large files/folders, so taking their log (i.e. their size in number of bits) makes their size evenly distributed.

<details><summary>I worked this out by</summary>

```
du -x -b --time=atime --time-style=+%s duke/projects | tee results
```

and plotting with matplotlib:

<details><summary>plot.py</summary>

```python
#!/usr/bin/env python3

import time
import numpy as np
import matplotlib.pyplot as plt

data = np.genfromtxt("results", delimiter="\t")
data = data.astype(int)

size, age = data[:,0], int(time.time()) - data[:,1]

plt.close();
ax=plt.gca();
ax.scatter(size, age/60/60/24);
ax.set_xscale('log', base=2);  # **on a log scale**
plt.title('Contents of /mnt/duke/projects')
plt.xlabel('size (bytes)');
plt.ylabel('age (days)');
plt.savefig("scatter.png")
plt.show()
```

</details>

![scatter](https://user-images.githubusercontent.com/987487/143777385-ab4118ca-a5d1-41b3-8c15-d0dacd8e8b19.png)

Then reflecting on the what the goal should be. I decided I would like to colour in the top-right corner of this scatter plot as targets to be rid of: they are the ones no one has touched in forever that are taking up the most space. The question is just how to trade off between them. Clearly the top-right corner itself is the highest priority, 

I realized after trying a few designs like 

![3d](https://user-images.githubusercontent.com/987487/143777524-d716dc16-e1e6-40e1-b4e3-c0455ac8c32a.png)

((yellow means `rank > 0.5` here))

that if I scaled both age and size to 0..1, then averaged, the final rank would also be scaled to 0..1 be, with 1 meaning very bad and 0 meaning no problem.

</details>

It would be more accurate to use

```
rank = (lg(size)/max(lg(size)) + age/max(age))/2
```

but that would require an extra scan and I'm impatient and I think we can assume the distribution won't change very much over the next year or so at least.

Anyway, here's the results. It took me about 7 minutes to scan `duke/projects`:

```
u108545@joplin:~$ du -x -b --time=atime --time-style=+%s duke/projects | awk -v NOW="$(date +%s)" 'BEGIN{OFS=FS="\t"} {print ((log($1)/log(2))/30+(NOW - $2)/60/60/24/900)/2, $1, $2, $3}' | sort -nr | head -n 50 | awk 'BEGIN{OFS=FS="\t"} {print $4}' | tr '\n' '\0' | xargs -0 -n 1 du --time=atime -h -d 0 |  tee /tmp/large-projects-atime.txt
du: cannot read directory 'duke/projects/archived/2017/Connectome/2013/Connectome/intersubject': Permission denied
du: cannot read directory 'duke/projects/archived/2017/Connectome/2013/Connectome/KS_HCP21': Permission denied
du: cannot read directory 'duke/projects/archived/2017/Connectome/2013/Connectome/KS_HCP28': Permission denied
du: cannot read directory 'duke/projects/archived/2017/Connectome/2013/Connectome/KS_HCP29': Permission denied
du: cannot read directory 'duke/projects/archived/2017/Connectome/2013/Connectome/KS_HCP31': Permission denied
du: cannot read directory 'duke/projects/archived/2017/Connectome/2013/Connectome/KS_HCP32': Permission denied
du: cannot read directory 'duke/projects/archived/2017/Connectome/2013/Connectome/KS_HCP33/dcm/10_ep2d_diff_0.8mm_b24k_G212_d11_D35_rise19_TE72': Permission denied
du: cannot read directory 'duke/projects/archived/2017/Connectome/2013/Connectome/KS_HCP33/dcm/11_ep2d_diff_0.8mm_b24k_G212_d11_D35_rise19_TE72': Permission denied
du: cannot read directory 'duke/projects/archived/2017/Connectome/2013/Connectome/KS_HCP33/dcm/12_ep2d_diff_0.8mm_b24k_G169_d14_D35_rise19_TE74': Permission denied
du: cannot read directory 'duke/projects/archived/2017/Connectome/2013/Connectome/KS_HCP33/dcm/13_ep2d_diff_0.8mm_b24k_G169_d14_D35_rise19_TE74': Permission denied
50G	2019-09-04 12:57	duke/projects/archived/2017/Connectome
du: cannot read directory 'duke/projects/archived/2017/Connectome/2013/Connectome/intersubject': Permission denied
du: cannot read directory 'duke/projects/archived/2017/Connectome/2013/Connectome/KS_HCP21': Permission denied
du: cannot read directory 'duke/projects/archived/2017/Connectome/2013/Connectome/KS_HCP28': Permission denied
du: cannot read directory 'duke/projects/archived/2017/Connectome/2013/Connectome/KS_HCP29': Permission denied
du: cannot read directory 'duke/projects/archived/2017/Connectome/2013/Connectome/KS_HCP31': Permission denied
du: cannot read directory 'duke/projects/archived/2017/Connectome/2013/Connectome/KS_HCP32': Permission denied
du: cannot read directory 'duke/projects/archived/2017/Connectome/2013/Connectome/KS_HCP33/dcm/10_ep2d_diff_0.8mm_b24k_G212_d11_D35_rise19_TE72': Permission denied
du: cannot read directory 'duke/projects/archived/2017/Connectome/2013/Connectome/KS_HCP33/dcm/11_ep2d_diff_0.8mm_b24k_G212_d11_D35_rise19_TE72': Permission denied
du: cannot read directory 'duke/projects/archived/2017/Connectome/2013/Connectome/KS_HCP33/dcm/12_ep2d_diff_0.8mm_b24k_G169_d14_D35_rise19_TE74': Permission denied
du: cannot read directory 'duke/projects/archived/2017/Connectome/2013/Connectome/KS_HCP33/dcm/13_ep2d_diff_0.8mm_b24k_G169_d14_D35_rise19_TE74': Permission denied
50G	2019-09-04 12:57	duke/projects/archived/2017
41G	2019-09-04 14:13	duke/projects/ivadomed/spineGeneric_201904111053/result
37G	2019-09-04 13:07	duke/projects/axondeepseg/20190326_Eliana
36G	2019-09-04 14:08	duke/projects/ivadomed/spineGeneric_201904040901/result
25G	2019-09-04 14:33	duke/projects/ml_sci_prognosis/20190411-Handseg&straight/resampled_seg_whandseg
25G	2019-09-04 14:31	duke/projects/ml_sci_prognosis/20190408-Data/resampled_seg_whandseg
22G	2019-09-04 12:38	duke/projects/archived/2017/Connectome/2014
21G	2019-09-04 12:54	duke/projects/archived/2017/Connectome/2015
20G	2019-09-04 13:05	duke/projects/axondeepseg/20190326_Eliana/data_raw
16G	2019-09-04 14:26	duke/projects/ml_sci_prognosis/20190408-Data/firstpass_hanseg
16G	2019-09-04 14:27	duke/projects/ml_sci_prognosis/20190408-Data/redo_handseg
15G	2019-09-04 14:01	duke/projects/HeadFix
15G	2019-09-04 14:01	duke/projects/HeadFix/Head_fix_data
15G	2019-09-04 15:56	duke/projects/ms_seg/sc_auto_seg
25G	2019-09-30 14:09	duke/projects/ml_sci_prognosis/20190408-Data/resampled
11G	2019-09-04 13:07	duke/projects/axondeepseg/20190326_Eliana/results
11G	2019-09-04 16:03	duke/projects/RAMRI/RAMRI_data/CantUse_data/RAMRI_075/Processed_data
9.7G	2019-09-04 13:03	duke/projects/axondeepseg/20190326_Eliana/data_raw/DMS images
9.2G	2019-09-04 15:58	duke/projects/ms_seg/seg_paper/data
17G	2019-09-29 16:03	duke/projects/ivadomed/olivier/models
9.0G	2019-09-04 16:04	duke/projects/RAMRI/RAMRI_data/RAMRI_data_asl/CBF/RAMRI_data_CBF_CD/data_conv
8.6G	2019-09-04 13:05	duke/projects/axondeepseg/20190326_Eliana/data_raw/samples_to_train_all
8.3G	2019-09-04 15:56	duke/projects/ms_seg/sc_auto_seg/sc_auto_seg
8.3G	2019-09-04 15:56	duke/projects/ms_seg/sc_auto_seg/sc_auto_seg/duke2rosenberg_171124
7.8G	2019-09-04 17:03	duke/projects/stainless_microscopy/Pics
7.6G	2019-09-04 17:00	duke/projects/stainless_microscopy/ISMRM2019_old
7.5G	2019-09-04 16:03	duke/projects/RAMRI/RAMRI_data/CantUse_data/RAMRI_075/Processed_data/Toad_wit_upsampling
7.5G	2019-09-04 17:00	duke/projects/stainless_microscopy/ISMRM2019_old/FV1000_space
7.0G	2019-09-04 15:51	duke/projects/ms_seg/c2c3
7.0G	2019-09-04 15:51	duke/projects/ms_seg/c2c3/data
6.3G	2019-09-04 13:00	duke/projects/axondeepseg/20180215_AxonDeepSeg_full_slice_segs
6.2G	2019-09-04 13:10	duke/projects/axondeepseg/20190417_Eliana/data
6.2G	2019-09-04 17:03	duke/projects/stainless_microscopy/Pics/Micro_FV1000
30G	2019-11-12 12:52	duke/projects/ms_brain_spine/segBrain_auto_tests
5.8G	2019-09-04 14:01	duke/projects/generic_spine_procotol
5.8G	2019-09-04 14:01	duke/projects/generic_spine_procotol/data
5.4G	2019-09-04 12:36	duke/projects/archived/2017/Connectome/2014/HC_SC_007
5.4G	2019-09-04 16:07	duke/projects/RAMRI/RAMRI_data/RAMRI_data_asl/CBF/RAMRI_data_CBF_CD/mask
4.9G	2019-09-04 15:59	duke/projects/ms_seg/seg_paper/data_ms
4.9G	2019-09-04 13:06	duke/projects/axondeepseg/20190326_Eliana/results/DMS_images_8bit
4.9G	2019-09-04 13:09	duke/projects/axondeepseg/20190417_Eliana/data/DMS_images_8bit
4.9G	2019-09-04 13:07	duke/projects/axondeepseg/20190326_Eliana/results/DMS_images_8bit_corrected
4.8G	2019-09-04 16:36	duke/projects/RAMRI/Study01/kernels/NODDI
4.8G	2019-09-04 16:36	duke/projects/RAMRI/Study01/kernels
4.8G	2019-09-04 15:54	duke/projects/ms_seg/sc_auto_seg/miccai
4.8G	2019-09-04 15:54	duke/projects/ms_seg/sc_auto_seg/miccai/duke2rosenberg_180109
4.8G	2019-09-04 16:07	duke/projects/RAMRI/RAMRI_data/RAMRI_data_asl/CBF/RAMRI_data_CBF_CD/data_dicom
8.7G	2019-10-02 10:07	duke/projects/ml_sci_prognosis/201908-t1:t2crop/crop_prognosis
4.5G	2019-09-04 13:17	duke/projects/archived/cat_sci
```

If we deleted most of these we'd free up something like 20% of duke's space.

**Comment by jcohenadad** [comment]
why not simply move projects/archived under another location archives/projects

IIRC this is what we decided to do in the past @alexfoias can chip in

**Comment by kousu** [comment]
That's a great first step. There's lots more that my script found that's not `projects/archived` there. They look like old intermediate processings and experimental trainings and the like. e.g.

* `duke/projects/axondeepseg/20190326_Eliana/results`
* `duke/projects/axondeepseg/20180215_AxonDeepSeg_full_slice_segs`
* `uke/projects/ml_sci_prognosis/20190411-Handseg&straight/resampled_seg_whandseg`

No one has even opened these files (that's what `atime` means: "access time") in almost 3 years.

**Comment by joshuacwnewton** [comment]
FWIW, I have one potential reason why we shouldn't delete these files, and instead move/archive them as @jcohenadad suggests.

Recently I've been following up on past NeuroPoly research efforts for the purposes of [this SCT PR](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3541). My goal was to better document these older models included in `sct_deepseg`, because many of these models have a very bare repository that doesn't describe the training data used (e.g. https://github.com/ivadomed/mice_uqueensland_gm/). 

Often, the only clue for dataset details is [a hardcoded link to a location on duke](https://github.com/ivadomed/mice_uqueensland_gm/blob/92ae56969ce5bc14c6d723701b072f92dffa40b3/mice_uqueensland_gm.json#L7):

```json
        "bids_path": "../duke/projects/ivado-medical-imaging/data_mice_abdullah-UQ/",
```

So, if a dataset like this is deleted, then it becomes impossible to retroactively document them. (And, these would be the kind of datasets that haven't been touched in 3+ years, as you mention.)

<details>
<summary> Aside... </summary> 

Now, you could say that the root issue here is the fact that the repositories weren't well documented in the first place. But... what's done is done, I suppose? I digress...

</details>

Basically, what I'm trying to say is... just because a dataset hasn't been touched in years doesn't mean that it's of no use to anyone in the future (even for these seemingly useless intermediate datasets).

So, that's why I'm leaning more towards preserving everything, and not spending the effort trying to programmatically figure out which datasets are safe to delete. (And, to that point, is it possible that the cost of storage drives is cheaper than the dev hours needed to write and debug such a script, anyway?)

**Comment by alexfoias** [comment]
> why not simply move projects/archived under another location archives/projects
> 
> IIRC this is what we decided to do in the past @alexfoias can chip in
> 
> another good reason for doing this is the HD requirement (no need for fast access if project is archived)

Yes, in the past we moved project to `archives/projects/XXXX` , where `XXXX` is the year.

Note that at this moment /archives is not backed up. This will be done once we have more disk space.

**Comment by kousu** [comment]
> https://github.com/ivadomed/mice_uqueensland_gm/blob/92ae56969ce5bc14c6d723701b072f92dffa40b3/mice_uqueensland_gm.json#L7
> ...
> Now, you could say that the root issue here is the fact that the repositories weren't well documented in the first place. But... what's done is done, and I feel like this sort of situation is hard to fully prevent, anyway!

Thank you for chiming in. That's a useful perspective we hadn't considered. Penultimately I *do* say they weren't done properly in the first place.

Would you go as far as `wontfix`ing this entire issue @joshuacwnewton? Just close it without cleaning things up?

The pressing issue is that while `duke` has a ton of space, it's also a ton of wasted space and our datasets are _large_:

```
root@joplin:~# df -h /mnt/duke/*
Filesystem                           Size  Used Avail Use% Mounted on
/dev/sda1                            440G  363G   55G  87% /
//duke.neuro.polymtl.ca/histology    9.0T  7.4T  1.7T  82% /mnt/duke/histology
//duke.neuro.polymtl.ca/mri          8.9T  4.4T  4.6T  50% /mnt/duke/mri
//duke.neuro.polymtl.ca/projects     4.4T  4.1T  343G  93% /mnt/duke/projects
//duke.neuro.polymtl.ca/public       4.4T  4.1T  343G  93% /mnt/duke/public
//duke.neuro.polymtl.ca/sct_testing  8.9T  4.4T  4.6T  50% /mnt/duke/sct_testing
//duke.neuro.polymtl.ca/temp         8.9T  4.4T  4.6T  50% /mnt/duke/temp
```

`/mnt/duke/{projects,public}` (they're actually the same disk) could easily be used up within the year by just a couple of graduate students. The msseg challenge dataset this summer produced something like 100G.

**Comment by joshuacwnewton** [comment]
> Would you go as far as `wontfix`ing this entire issue @joshuacwnewton? Just close it without cleaning things up?

It depends on how pressing our storage issues are. If we can afford to preserve the existing datasets, then I'd want to do so, I think?

But, I don't mean to suggest that we shouldn't try to clean up things ever, though! I'm just thinking that it would be safer and easier to target present/future datasets, rather than retroactively targeting 3+ year old datasets. 

> The msseg challenge dataset this summer produced something like 100G.

In a case like this, I'm wondering if we can check in with the students as the project is happening, so that we make sure that the only datasets that get preserved are ones that are directly tied to some sort of hardcoded link/commit/presentation/paper/etc.

(Which it does seem like we're doing, what with the `git annex` efforts and the versioned datasets and what have you, right? I can only imagine that if we were to do-over all of the `sct_deepseg` models that I mentioned, then those datasets would have been preserved on `data.neuro.polymtl.ca` had it existed at the time, no?)

**Comment by kousu** [comment]
> But, I don't mean to suggest that we shouldn't try to clean up things ever, though! I'm just thinking that it would be safer and easier to target present/future datasets with cleanup efforts, rather than retroactively targeting 3+ year old datasets (since at that time, our data policies may have been a bit looser, with important datasets mixed in with useless ones).

How hard do you think it would be to do a giant `grep` of every old version of every repo we have for the word `duke`? Maybe we could see if we're talking about like...100s of folders or maybe like 7 that we need to preserve.

**Comment by joshuacwnewton** [comment]
> How hard do you think it would be to do a giant `grep` of every old version of every repo we have for the word `duke`? Maybe we could see if we're talking about like...100s of folders or maybe like 7 that we need to preserve.

Personally, my gut says to start from a position of "preserve everything", and only spend the effort to analyze the various datasets once we get to a point where preserving everything stops being feasible.

To answer your question, though: It's probably possible! It's just a question of whether doing that is worth the effort (vs. simply moving it to `archives`, putting it out of our minds, and moving on).

**Comment by jcohenadad** [comment]
sorry i don’t have time to chip in with long discussion post— i’d like to discuss this IRL— a lot of content raised in this post needs background info/context— IRL meeting wl be more efficient

**Comment by kousu** [comment]
This is a duplicate of https://github.com/neuropoly/computers/issues/185

**Comment by namgo** [comment]
This is the next (by order of important) issue in my task list, I don't have the access to work on it right now, however I want to get a sense of what needs to happen to resolve it.

I'm reading through comments from 2021 and I don't know how relevant any given point made is.

What progress has been made? What are the next steps?

This is listed as a dependency for https://github.com/neuropoly/computers/issues/443 and https://github.com/neuropoly/computers/issues/425

**Comment by jcohenadad** [comment]
hi Nathan-- this one is tough-- essentially i/we/students need to remove files/folders that should not be here-- sometimes students create files/folders for temporary purposes and cleanup needs to be done-- i think i should be involved in this as i am overseeing all projects in the lab-- some things should be deleted, some should not-- i wish there was a better system
